pkg/sql/resolver.go

```diff
@@ -156,6 +161,12 @@ func (p *planner) ResolveType(
 	}
 	tn := tree.MakeTypeNameFromPrefix(prefix, tree.Name(name.Object()))
 	tdesc := desc.(*sqlbase.ImmutableTypeDescriptor)
+
+	if p.databaseContext != sqlbase.InvalidID && tdesc.ParentID != sqlbase.InvalidID && tdesc.ParentID != p.databaseContext {
```

Comment:
when does parentid != invalidid, or dbcontext != invalidid?

Reply:
`parentID != invalidID` when the resolved descriptor is a real type descriptor (not a builtin type referenced with a virtual schema prefix). `databaseContext != invalidID` when it is set through `planner.RunWithOptions`. I'll update this to include a comment about that.